library/core/src/mem/maybe_uninit.rs

```diff
@@ -726,15 +726,30 @@ impl<T> MaybeUninit<T> {
     /// `assume_init_read` and then [`assume_init`]), it is your responsibility
     /// to ensure that data may indeed be duplicated.
     ///
+    /// Additionally, if you call this function from a different thread than the one
```

Comment:
It would be helpful for this paragraph to start with a sentence that clearly identifies the problem. I would recommend:

> Note that this function does *not* require `T: Send`. If you call this function from a different thread...

Reply:
Exactly. I thought for a bit, and what do you think about:

> Since this function does *not* require `T: Send`, you also need to carefully treat thread safety. If you call this function...

Reply:
Well, your point seems to have a reason. To be exact, I understand:

> I prefer my original wording. I don't think "carefully treat thread safety" adds any value

but I'm having a hard time parsing the last clause:

> it's already implicit from obeying the requirements of the Send trait.

It's not like using this method cross-thread to non-`Send` `T` is an immediate UB, and "Note that this function does not require `T: Send`" doesn't seem to "obey" the requirements of the `Send` trait, anyway. 

Sorry, I know my English is hard to read here, but I'm having, like, UB from semantical parse failure. I'm not saying I'm a bot, but it's triggering UB to me, likely due to my limited English fluency.

Reply:
(BTW, I don't re-claim but do note that I thought the small "value" of the part is *explicitly* title the paragraph that this part is about thread safety. I do accept your claim if it is that it's implicit and obvious from the shorter version, but I'm not sure if I understand it well)

Reply:
Good point, disregard my last comment then. I would only make one change to the wording, replace "treat" with "consider" since that is more correct in this context.